### ISSUE TITLE
[TypeScript Resolvers] union type resolvers aren't generic

### ISSUE THREAD

**Issue by e-r-w** [dotansimha/graphql-code-generator#1197]
**Describe the bug**
When a schema contains union types, the generated resolvers don't contain the generics that regular types do. This wouldn't be so much of a problem except the `IResolvers` generated interface seem to expect all it's members to be generics in order to pass on the `Context` type argument

**To Reproduce**
Given a schema:
```graphql
type Foo {
  a: String!
}

type Bar {
  b: String!
}

union FooBar = Foo | Bar

type Query {
  foobars: FooBar!
}

schema {
  query: FooBar
}
```

and using the typescript resolver plugin, we get:

```typescript
export namespace FooResolvers {
  // This is fine, accepts the context type argument
  export interface Resolvers<Context = {}, TypeParent = Foo> {
    a?: AResolver<string, TypeParent, Context>;
  }

  export type AResolver<R = string, Parent = Foo, Context = {}> = Resolver<
    R,
    Parent,
    Context
  >;
}

//...

export namespace FooBarResolvers {
  // Does not accept type arguments
  export interface Resolvers {
    __resolveType: ResolveType;
  }
  export type ResolveType<
    R = "Foo" | "Bar",
    Parent = Foo | Bar,
    Context = {}
  > = TypeResolveFn<R, Parent, Context>;
}

// ...

export interface IResolvers<Context = {}> {
  Foo?: FooResolvers.Resolvers<Context>;
  Bar?: BarResolvers.Resolvers<Context>;
  Query?: QueryResolvers.Resolvers<Context>;
  FooBar?: FooBarResolvers.Resolvers<Context>;
// Type 'Resolvers' is not generic. [2315]
}
```

This causes a compilation error in Typescript v3.2.1

**Expected behavior**
Not sure if `IResolvers` props for union types should not be generic, or that union type resolvers should be generic in order to pass the type arguments through, e.g:

```typescript
export namespace FooBarResolvers {
  export interface Resolvers<
    R = "Foo" | "Bar",
    Parent = Foo | Bar,
    Context = {}
  > {
    __resolveType: ResolveType<R, Parent, Context>;
  }
  export type ResolveType<
    R = "Foo" | "Bar",
    Parent = Foo | Bar,
    Context = {}
  > = TypeResolveFn<R, Parent, Context>;
}
```

Either one would work fine although the latter would be more complete types IMHO

**Environment:**
 - OS: OSX 10.13.3
 - Codegen: 0.16.0
 - Node: 11

**Comment by e-r-w** [comment]
Just noticed #1194 so closing this issue

**Comment by dotansimha** [comment]
Hi @e-r-w ! Reopened so we can track this for the next release.

**Comment by dotansimha** [comment]
Fixed in 0.16.1 :)
